From 910de2ecdbf1c329ad1b926db49da325c0114047 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Fri, 17 Apr 2020 07:35:01 -0500
Subject: [PATCH 19/21] arm64: dts: beacon-imx8mm: Update PMIC entries to match
 backported driver

The driver was backported from 5.4 kernel, so this patch updates the
device tree entry to match that of the 8MM EVK PMIC swttings from
which the Beacon board was based.

sing the following command, I can determine the state and voltage of all the known regulators:

    cat /sys/kernel/debug/regulator/regulator_summary

DVFS seems to be working on the software side with a software update:

1.2GHz:

    BUCK2               0    1      0   850mV     0mA   700mV  1300mV
        cpu0                                    850mV   850mV

1.6GHz

    BUCK2              0    1      0   950mV     0mA   700mV  1300mV
        cpu0                                     950mV   950mV

1.8GHz

    BUCK2              0    1      0  1000mV     0mA   700mV  1300mV
        cpu0                                                         1000mV  1000mV

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 .../boot/dts/freescale/beacon-imx8mm-som.dtsi | 94 +++++++------------
 1 file changed, 33 insertions(+), 61 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
index b15521b9767b..583f572b000a 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
@@ -61,27 +61,17 @@
 	pinctrl-0 = <&pinctrl_i2c1>;
 	status = "okay";
 
-	pmic: bd71837@4b {
+	pmic@4b {
 		reg = <0x4b>;
-		compatible = "rohm,bd71840", "rohm,bd71837";
-		/* PMIC BD71837 PMIC_nINT GPIO1_IO3 */
+		compatible = "rohm,bd71847";
 		pinctrl-0 = <&pinctrl_pmic>;
-		gpio_intr = <&gpio1 3 GPIO_ACTIVE_LOW>;
-
-		gpo {
-			rohm,drv = <0x0C>;	/* 0b0000_1100 all gpos with cmos output mode */
-		};
+		interrupt-parent = <&gpio1>;
+		interrupts = <3 GPIO_ACTIVE_LOW>;
+		rohm,reset-snvs-powered;
 
 		regulators {
-			#address-cells = <1>;
-			#size-cells = <0>;
-
-			bd71837,pmic-buck2-uses-i2c-dvs;
-			bd71837,pmic-buck2-dvs-voltage = <1000000>, <900000>, <0>; /* VDD_ARM: Run-Idle */
-
-			buck1_reg: regulator@0 {
-				reg = <0>;
-				regulator-compatible = "buck1";
+			buck1_reg: BUCK1 {
+				regulator-name = "BUCK1";
 				regulator-min-microvolt = <700000>;
 				regulator-max-microvolt = <1300000>;
 				regulator-boot-on;
@@ -89,105 +79,87 @@
 				regulator-ramp-delay = <1250>;
 			};
 
-			buck2_reg: regulator@1 {
-				reg = <1>;
-				regulator-compatible = "buck2";
+			buck2_reg: BUCK2 {
+				regulator-name = "BUCK2";
 				regulator-min-microvolt = <700000>;
 				regulator-max-microvolt = <1300000>;
 				regulator-boot-on;
 				regulator-always-on;
 				regulator-ramp-delay = <1250>;
+				rohm,dvs-run-voltage = <1000000>;
+				rohm,dvs-idle-voltage = <900000>;
 			};
 
-			buck3_reg: regulator@2 {
-				reg = <2>;
-				regulator-compatible = "buck3";
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1300000>;
-			};
-
-			buck4_reg: regulator@3 {
-				reg = <3>;
-				regulator-compatible = "buck4";
-				regulator-min-microvolt = <700000>;
-				regulator-max-microvolt = <1300000>;
-			};
-
-			buck5_reg: regulator@4 {
-				reg = <4>;
-				regulator-compatible = "buck5";
+			buck3_reg: BUCK3 {
+				// BUCK5 in datasheet
+				regulator-name = "BUCK3";
 				regulator-min-microvolt = <700000>;
 				regulator-max-microvolt = <1350000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			buck6_reg: regulator@5 {
-				reg = <5>;
-				regulator-compatible = "buck6";
+			buck4_reg: BUCK4 {
+				// BUCK6 in datasheet
+				regulator-name = "BUCK4";
 				regulator-min-microvolt = <3000000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			buck7_reg: regulator@6 {
-				reg = <6>;
-				regulator-compatible = "buck7";
+			buck5_reg: BUCK5 {
+				// BUCK7 in datasheet
+				regulator-name = "BUCK5";
 				regulator-min-microvolt = <1605000>;
 				regulator-max-microvolt = <1995000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			buck8_reg: regulator@7 {
-				reg = <7>;
-				regulator-compatible = "buck8";
+			buck6_reg: BUCK6 {
+				// BUCK8 in datasheet
+				regulator-name = "BUCK6";
 				regulator-min-microvolt = <800000>;
 				regulator-max-microvolt = <1400000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo1_reg: regulator@8 {
-				reg = <8>;
-				regulator-compatible = "ldo1";
+			ldo1_reg: LDO1 {
+				regulator-name = "LDO1";
 				regulator-min-microvolt = <3000000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo2_reg: regulator@9 {
-				reg = <9>;
-				regulator-compatible = "ldo2";
+			ldo2_reg: LDO2 {
+				regulator-name = "LDO2";
 				regulator-min-microvolt = <900000>;
 				regulator-max-microvolt = <900000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo3_reg: regulator@10 {
-				reg = <10>;
-				regulator-compatible = "ldo3";
+			ldo3_reg: LDO3 {
+				regulator-name = "LDO3";
 				regulator-min-microvolt = <1800000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo4_reg: regulator@11 {
-				reg = <11>;
-				regulator-compatible = "ldo4";
+			ldo4_reg: LDO4 {
+				regulator-name = "LDO4";
 				regulator-min-microvolt = <900000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-boot-on;
 				regulator-always-on;
 			};
 
-			ldo6_reg: regulator@13 {
-				reg = <13>;
-				regulator-compatible = "ldo6";
+			ldo6_reg: LDO6 {
+				regulator-name = "LDO6";
 				regulator-min-microvolt = <900000>;
 				regulator-max-microvolt = <1800000>;
 				regulator-boot-on;
-- 
2.17.1

